### PR TITLE
feat: Dockerize demo projects

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Sorry all, this is a big one.

This PR Dockerizes all the projects in this repo, and uses Docker Compose to make building and starting multiple projects locally a bit easier.

I tried to keep a light touch with the individual projects. Most updates to projects are to fix issues with Rollup that I ran into, or to fix a bug after recent package version bumps from Renovate.

If it's easier, I can create new PRs for those project fixes. I think there were just one or two, one being the update to [`server.js`](https://github.com/freeCodeCamp/demo-projects/blob/0dee2a71f31c0355f780aedc07f87b55b1994b87/apps/build-a-pinterest-clone/server.js) for the Build a Pinterest Clone project.

Another big change was tweaking the method to refresh the access token for the Twitch Proxy project and moving those scripts into that project's directory. That project's container will also need to be set up to restart when it fails so that it can fetch a new access token every month, but that will probably depend on how the project is deployed.

Finally, the tests were simplified, and there's a new test to check the status code for every project.

Edit: These commits are all over the place, and I'd be happy to squash them if that makes things easier :)

### CODE REVIEW

File: apps/build-a-pinterest-clone/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# Copy over all the files in the project directory to /app early
+# for rollup bundling
+COPY . . 
+
+ENV PORT=3000
+ENV GITHUB_KEY=${BUILD_A_PINTEREST_CLONE_GITHUB_KEY}
+ENV GITHUB_SECRET=${BUILD_A_PINTEREST_CLONE_GITHUB_SECRET}
+ENV DB_URI=${BUILD_A_PINTEREST_CLONE_DB_URI}
+ENV APP_URL=${BUILD_A_PINTEREST_CLONE_APP_URL}
```
Comment (ShaunSHamilton) [inline apps/build-a-pinterest-clone/Dockerfile:13]: Why do we repeat all the env variables in each Dockerfile and `.env` file?

Comment (raisedadead) [reply apps/build-a-pinterest-clone/Dockerfile:13]: That's to make the "container deployment" platform agnostic.

1. Docker Compose in a single VM - Use a `.env` file and startup the apps.
2. Azure App Service / AWS Fargate or something similar - just set the respective values in the UI and point it to the registry and the image.

It looks a bit ugly code-wise, but that's worth the flexibility.

File: apps/25--5-clock/client/index.jsx [hunk 2]
```diff
@@ -257,7 +257,7 @@ class Timer extends React.Component {
         <div className='author'>
           {' '}
           Designed and Coded by <br />
-          <a href='https://goo.gl/6NNLMG' target='_blank'>
+          <a href='https://goo.gl/6NNLMG' target='_blank' rel='noreferrer'>
```
Comment (raisedadead) [inline apps/25--5-clock/client/index.jsx:260]: I think we do not want to use short links (regardless of the intent). @no-stack-dub-sack if you are around, can you confirm if we should swap this with a more transparent link?

Comment (raisedadead) [reply apps/25--5-clock/client/index.jsx:260]: Yes, please.

Comment (no-stack-dub-sack) [reply apps/25--5-clock/client/index.jsx:260]: @raisedadead long time no talk! Hope you're well, man! 

Fwiw, I have no idea whether I added these short links way back when, or if they were added by someone else along the way, but in either case, I agree that links should be more transparent.

Additionally, please feel free to remove any attributions to me like this. I don't think I can claim any credit for this anymore. These projects have been touched by so many people over the years that I'm hardly the author of any of this at all! Very cool to see how its all evolved and that this is actually still around, and being used and actively developed some 5-6 years later :-)

Comment (raisedadead) [reply apps/25--5-clock/client/index.jsx:260]: Thanks, Peter.

Yes - it's been a while. Lol.

We will take care of the links. Everything else can stay. These projects have a legacy and nostalgia of their own. Thanks to early contributors like yourself, we have over 10 million users who find these resources helpful.

We obviously could not have achieved this without your help :)

BTW, we have a lot in the works too.

Thanks again.

**Comment by scissorsneedfoodtoo** [comment]
Note: The last commit bumps the Node version for all containers to v18, and requires Node v18 and npm v9 in the main `package.json` file.

**Comment by raisedadead** [comment]
Once you are done, do not forget to clean up with the following:

```console
docker compose down
docker system prune --volumes --all --force
```

**Pull request merged**